appinventor/appengine/src/com/google/appinventor/client/editor/simple/components/MockVisibleExtension.java

```diff
@@ -0,0 +1,281 @@
+package com.google.appinventor.client.editor.simple.components;
+
+import com.google.appinventor.client.Ode;
+import com.google.appinventor.client.editor.simple.SimpleEditor;
+import com.google.appinventor.client.utils.Promise;
+import com.google.appinventor.client.utils.ShadowRoot;
+import com.google.appinventor.client.utils.jstypes.*;
+import com.google.appinventor.client.utils.jstypes.Worker.MessageEvent;
+import com.google.appinventor.client.widgets.properties.EditableProperty;
+import com.google.appinventor.shared.simple.ComponentDatabaseInterface.MockInfo;
+import com.google.gwt.core.client.JsonUtils;
+import com.google.gwt.dom.client.Document;
+import com.google.gwt.dom.client.Element;
+import com.google.gwt.json.client.JSONBoolean;
+import com.google.gwt.json.client.JSONNumber;
+import com.google.gwt.json.client.JSONObject;
+import com.google.gwt.json.client.JSONString;
+import com.google.gwt.json.client.JSONValue;
+import com.google.gwt.user.client.Window;
+import com.google.gwt.user.client.ui.FlowPanel;
+import com.google.gwt.user.client.ui.HTMLPanel;
+import com.google.gwt.user.client.ui.Image;
+import com.google.appinventor.components.common.PropertyTypeConstants;
+
+public class MockVisibleExtension extends MockVisibleComponent {
+  private final String MOCK_RENDER_REQUEST = "render-request";
+
+  private final long projectId;
+  private final String packageName;
+  private final MockInfo mockInfo;
+
+  private Worker worker;
+  private String workerUrl;
+
+  private final ShadowRoot shadowRoot;
+
+  public MockVisibleExtension(
+      SimpleEditor editor,
+      String typeName,
+      Image iconImage,
+      String packageName,
+      MockInfo mockInfo) {
+    super(editor, typeName, iconImage);
+    Ode.CLog("MockVisibleExtension.constructor");
```

Comment:
nit: we should remove these log statements before merging.

Reply:
Yes, this and a bunch of others need to go.